### PR TITLE
LoggerManager, CredentialsManager, APIManager classes

### PR DESCRIPTION
**Pull request by Agustin-Mediotti**

### CODE REVIEW

File: src/main/java/api/APIManager.java [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+
+    public static APIManager getInstance() {
+        return (instance == null) ? instance = new APIManager() : instance;
+    }
+
+    private void initialize() {
+        log.info("Initializing API Manager");
+        RestAssured.baseURI = CredentialsManager.getInstance().getBaseUrl();
+        RestAssured.basePath = CredentialsManager.getInstance().getBasePath();
+    }
```
Comment (eliosfg) [inline src/main/java/api/APIManager.java:28]: You have defined port 80 in the api.properties file, but you haven't set that property here

**Review by eliosfg — CHANGES_REQUESTED** [review]
See may comment

File: src/main/java/api/APIManager.java (hunk 1 above)
Comment (Agustin-Mediotti) [reply src/main/java/api/APIManager.java]: Resolved in last commit

**Review by adrianovi — APPROVED** [review]

File: src/main/java/api/APIManager.java [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+        RestAssured.basePath = CredentialsManager.getInstance().getBasePath();
+        RestAssured.port = CredentialsManager.getInstance().getAPIServicePort();
+    }
+
+    public void setCredentials(String username, String password) {
+        PreemptiveBasicAuthScheme authScheme = new PreemptiveBasicAuthScheme();
+        authScheme.setUserName(username);
+        authScheme.setPassword(password);
+        RestAssured.authentication = authScheme;
+    }
```
Comment (eliosfg) [inline src/main/java/api/APIManager.java:36]: This authentication type won't be necessary for the WordPress API.

[the remark this answers is not in the document]
Comment (Agustin-Mediotti) [reply src/main/java/api/APIManager.java]: Done, good review! 🧉

**Pull request merged**